projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js

```diff
@@ -244,6 +245,77 @@ async function checkForEscalation( issueReferences, commentBody, escalationNote,
 	return true;
 }
 
+/**
+ * Add or update a label on the issue to indicate a number range of support references,
+ * once it has gathered more than 10 support references.
+ *
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {string} repo - Repository name.
+ * @param {string} ownerLogin - Owner of the repository.
+ * @param {number} number - Issue number.
+ * @param {number} issueReferencesCount - Number of support references gathered in this issue.
+ * @returns {Promise<void>}
+ */
+async function addOrUpdateInteractionCountLabel(
+	octokit,
+	repo,
+	ownerLogin,
+	number,
+	issueReferencesCount
+) {
+	const ranges = [ 50, 20, 10 ];
+
+	// Check if our issue has issues in one of the ranges where we want to label it.
+	const issueRange = ranges.find( range => issueReferencesCount > range );
+
+	// Bail if our issue hasn't gathered enough support references to warrant a label.
+	if ( ! issueRange ) {
+		return;
+	}
+
+	// Name of the label we want to add to this issue.
+	const interactionCountLabel = `[Interaction #] > ${ issueRange }`;
+
+	debug(
+		`gather-support-references: Issue #${ number } has gathered ${ issueReferencesCount } support references. It deserves a label, "${ interactionCountLabel }"`
+	);
+
+	// Check if the issue already has this label.
+	const labels = await getLabels( octokit, ownerLogin, repo, number );
+	const existingInteractionCountLabels = labels.filter( label =>
+		label.startsWith( '[Interaction #]' )
+	);
+
+	// Our issue already has at least one interaction count label. Is is the right one?
+	if ( existingInteractionCountLabels.length > 0 ) {
+		await Promise.all(
+			existingInteractionCountLabels.map( async existingInteractionCountLabel => {
+				// if that's not the one we want to add, remove it.
+				if ( existingInteractionCountLabel !== interactionCountLabel ) {
+					debug(
+						`gather-support-references: Issue #${ number } already has a label, "${ existingInteractionCountLabel }". We want to add the "${ interactionCountLabel }" label. Removing the "${ existingInteractionCountLabel }" label.`
+					);
+					await octokit.rest.issues.removeLabel( {
+						owner: ownerLogin,
+						repo,
+						issue_number: +number,
+						name: existingInteractionCountLabel,
+					} );
+				}
+			} )
+		);
+	}
+
+	// Add the label to our issue.
+	debug( `gather-support-references: Adding the "${ interactionCountLabel }" label.` );
+	await octokit.rest.issues.addLabels( {
+		owner: ownerLogin,
+		repo,
+		issue_number: +number,
+		labels: [ interactionCountLabel ],
+	} );
```

Comment:
This should be wrapped in a `if ( ! existingInteractionCountLabels.includes( interactionCountLabel ) )` test or something similar. No point in adding the label if it's already there.

Reply:
Good point. Done in 0a5184a8956a4b200237da1b5acabbd47fea07ad